src/buffer_manager.ts

```diff
@@ -638,6 +633,17 @@ export class BufferManager implements Disposable {
         actions.fireNvimEvent("document_buffer_init", bufId);
     }
 
+    private async bufName(docUri: Uri): Promise<string> {
+        if (docUri.scheme !== "file") {
+            return docUri.toString();
+        } else if (config.useWsl) {
+            const wslpath = await actions.lua<string>("wslpath", docUri.fsPath);
+            return wslpath;
```

Comment:
Can't we set generic types on the original code?

Reply:
Yeah I guess so. I just found the nested ternary hard to read and I did this to figure out what the problem was. But if you prefer the ternary I can do that.

Reply:
The ternary operator is difficult to read when not formatted. 😂